### PR TITLE
Update cfg-if to v1.10

### LINKED ISSUE #30: Is there a reason why `cfg-if` is locked to  `0.1.9`?
```toml
[dependencies]
cfg-if = "=0.1.9"
```

I'm getting an error like:

```log
error: failed to select a version for `cfg-if`.
<..snip..>
versions that meet the requirements `^0.1.10` are: 0.1.10
```

due to one of my dep library is using `cfg-if 0.1.10` while `directories-next` is locked to `0.1.9` version.

### PR DESCRIPTION
**Pull request by burtonageo**
Closes #30

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -18,7 +18,7 @@ keywords    = ["xdg", "basedir", "app_dirs", "path", "folder"]
 maintenance = { status = "passively-maintained" }
 
 [dependencies]
-cfg-if = "=0.1.9"
+cfg-if = "=0.1.10"
```
Comment (lzutao) [inline Cargo.toml:21]:
```suggestion
cfg-if = "0.1.10"
```

File: directories/Cargo.toml [hunk 2]
```diff
@@ -18,7 +18,7 @@ keywords    = ["xdg", "basedir", "app_dirs", "path", "folder"]
 maintenance = { status = "passively-maintained" }
 
 [dependencies]
-cfg-if = "=0.1.9"
+cfg-if = "=0.1.10"
```
Comment (lzutao) [inline directories/Cargo.toml:21]:
```suggestion
cfg-if = "0.1.10"
```

**Comment by XAMPPRocky** [comment]
FYI, unpinning `cfg-if`'s version is required for `tokei` to switch to `dirs-next` as the current pinning is too strict and causes a compile error where `dirs` currently doesn't.

File: CHANGELOG.md [hunk 3]
```diff
@@ -25,6 +25,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * `Fixed` for any bug fixes.
 * `Security` in case of vulnerabilities.
  -->
+## Unreleased
+
+* Changed: Update `cfg-if` dependency to `1.10`.
```
Comment (lzutao) [inline CHANGELOG.md:30]:
```suggestion
* Changed: Relax pinning `cfg-if` at `0.1.9`. Previous we inherit that from upstream repository. 
  The original author wanted to keep minimum Rust version at 1.13 which we don't anymore.
```

File: directories/CHANGELOG.md [hunk 4]
```diff
@@ -25,6 +25,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * `Fixed` for any bug fixes.
 * `Security` in case of vulnerabilities.
  -->
+ ## Unreleased
+
+* Changed: Update `cfg-if` dependency to `1.10`.
```
Comment (lzutao) [inline directories/CHANGELOG.md:30]: also this.

Comment (lzutao) [reply directories/CHANGELOG.md:30]:
```suggestion
### Fixed
* Relax pinning `cfg-if` at `0.1.9`. Previously we inherited that from upstream repository when forking.
  The original author wanted to keep minimum Rust version at 1.13 which we don't anymore.
```

**Comment by lzutao** [comment]
Continued in #32

**Pull request closed without merging**